controllers/activemqartemis_reconciler.go

```diff
@@ -1152,22 +1154,37 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) ProcessResources(customResource
 
 		for index := range delta.Added {
 			resourceToAdd := delta.Added[index]
-			reconciler.createResource(customResource, client, scheme, resourceToAdd, resourceType, reqLogger)
+			trackError(&compositeError, reconciler.createResource(customResource, client, scheme, resourceToAdd, resourceType, reqLogger))
 		}
 		for index := range delta.Updated {
 			resourceToUpdate := delta.Updated[index]
-			reconciler.updateResource(customResource, client, scheme, resourceToUpdate, resourceType, reqLogger)
+			trackError(&compositeError, reconciler.updateResource(customResource, client, scheme, resourceToUpdate, resourceType, reqLogger))
 		}
 		for index := range delta.Removed {
 			resourceToRemove := delta.Removed[index]
-			reconciler.deleteResource(customResource, client, scheme, resourceToRemove, resourceType, reqLogger)
+			trackError(&compositeError, reconciler.deleteResource(customResource, client, scheme, resourceToRemove, resourceType, reqLogger))
 		}
 	}
 
 	//empty the collected objects
 	reconciler.requestedResources = nil
 
-	return nil
+	if len(compositeError) == 0 {
+		return nil
+	} else {
+		// maybe errors.Join in go1.20
+		// using %q(uote) to keep errors separate
+		return fmt.Errorf("%q", compositeError)
```

Comment:
The function [errors.Join](https://pkg.go.dev/errors@master#Join) could be used to create an error that wraps the given errors. The error formats as the concatenation of the strings obtained by calling the Error method of each element of errs, with a newline between each string.

Reply:
not sure about having new lines in a condition message, but also join is in go1.20

Reply:
A newline between each string could be a good separator but I don't know if the Kubernetes/OpenShift console renders the new lines in a condition message.